scripts/releaseChangelog.js

```diff
@@ -0,0 +1,243 @@
+   */
+  const timeline = octokit.paginate.iterator(
+    octokit.repos.compareCommits.endpoint.merge({
+      owner: GIT_ORGANIZATION,
+      repo: GIT_REPO,
+      base: lastRelease,
+      head: release,
+    }),
+  );
+
+  /**
+   * @type {Octokit.ReposCompareCommitsResponseCommitsItem[]}
+   */
+  const commitsItems = [];
+  for await (const response of timeline) {
+    const { data: compareCommits } = response;
+    commitsItems.push(...compareCommits.commits.filter(filterCommit));
+  }
+
+  // Fetch all the pull Request and check if there is a section named changelog
+
+  const changeLogMessages = [];
+  await Promise.all(
+    commitsItems.map(async (commitsItem) => {
+      const searchPullRequestId = commitsItem.commit.message.match(/\(#([0-9]+)\)/);
+      if (!searchPullRequestId || !searchPullRequestId[1]) {
+        return;
+      }
+
+      const {
+        data: { body: bodyMessage },
+      } = await octokit.request('GET /repos/{owner}/{repo}/pulls/{pull_number}', {
+        owner: GIT_ORGANIZATION,
+        repo: GIT_REPO,
+        pull_number: Number(searchPullRequestId[1]),
+      });
+
+      if (!bodyMessage) {
+        return;
+      }
+
+      const changelogMotif = '# changelog';
+      const changelogIndex = bodyMessage.toLowerCase().indexOf(changelogMotif);
+      if (changelogIndex >= 0) {
+        changeLogMessages.push(
+          `From https://github.com/${GIT_ORGANIZATION}/${GIT_REPO}/pull/${
+            searchPullRequestId[1]
+          }\n${bodyMessage.slice(changelogIndex + changelogMotif.length)}`,
+        );
+      }
+    }),
+  );
+
+  // Get all the authors of the release
+  const authors = Array.from(
+    new Set(
+      commitsItems.map((commitsItem) => {
+        return commitsItem.author.login;
+      }),
+    ),
+  );
+
+  // Dispatch commits in different sections
+  const changeCommits = [];
+  const coreCommits = [];
+  const docsCommits = [];
+  const otherCommits = [];
+
+  commitsItems.forEach((commitItem) => {
+    const tag = parseTags(commitItem.commit.message);
+    switch (tag) {
+      case 'DataGrid':
+      case 'DataGridPro':
+      case 'l10n':
+      case '118n':
+        changeCommits.push(commitItem);
+        break;
+      case 'docs':
+        docsCommits.push(commitItem);
+        break;
+      case 'core':
+        coreCommits.push(commitItem);
+        break;
+      default:
+        otherCommits.push(commitItem);
+        break;
+    }
+  });
+
+  // Helper to print a list of commits in a section of the changelog
+  const logChangelogSection = (commitsList, header) => {
+    if (commitsList.length === 0) {
+      return '';
+    }
+
+    const sortedCommits = commitsList.sort((a, b) => {
+      const aTags = parseTags(a.commit.message);
+      const bTags = parseTags(b.commit.message);
+      if (aTags === bTags) {
+        return a.commit.message < b.commit.message ? -1 : 1;
+      }
+      return aTags.localeCompare(bTags);
+    });
+    return `${header}
```

Comment:
It should not add the header if it's an empty string. It adds unnecessary line breaks before inserting the other commits.

<img width="500" src="https://user-images.githubusercontent.com/42154031/149566190-e3201488-0260-480a-90b9-560dcb1f0afc.png" alt="image" style="max-width: 100%;">